submarine-workbench/workbench-server/src/test/java/org/apache/submarine/server/WorkbenchGitHubServerTest.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.submarine.server;
+
+import com.google.common.collect.Lists;
+import com.google.gson.Gson;
+import org.eclipse.jgit.api.PullResult;
+import org.eclipse.jgit.dircache.DirCache;
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.transport.PushResult;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.apache.submarine.database.utils.HttpRequestUtil.sendHttpRequest;
+
+public class WorkbenchGitHubServerTest {
+
+  private static final Logger LOG = LoggerFactory.getLogger(WorkbenchGitHubServerTest.class);
+  private static final String OWNER = "submarine-thirdparty";
+  private static final String REPO = "submarine_git_test";
+  WorkbenchGitHubServer workbenchGitHubServer = new WorkbenchGitHubServer();
+  private static final String REMOTE_PATH = "https://github.com/" + OWNER + "/" + REPO + ".git";
+  private static final String BRANCHNAME = "Branch-1";
+  private static final String LOCALPATH =
```

Comment:
Because everyone is using the same test repo, you need to add a random number to `LOCALPATH`.

Reply:
Done.